crates/sound/src/audio_backends/pw_backend.rs

```diff
@@ -383,4 +383,33 @@ impl AudioBackend for PwBackend {
 
         Ok(())
     }
+
+    fn release(&self, stream_id: u32) -> Result<()> {
+        println!("pipewire backend, release function");
+        self.thread_loop.lock();
+        let mut stream_states = self.stream_states.write().unwrap();
+        let stream_hash = self.stream_hash.read().unwrap();
+        let stream_listener = self.stream_listener.read().unwrap();
+
+        if let Some(stream) = stream_hash.get(&stream_id) {
+            println!("todo something with stream for release");
+            stream.disconnect().expect("could not disconnect stream");
+            //todo: destroy stream?
+            //see related issue in pipewire:rs  https://gitlab.freedesktop.org/pipewire/pipewire-rs/-/issues/53
+        } else {
+            println!("stream not found for the given stream_id");
+        };
+        if let Some(stream_lis) = stream_listener.get(&stream_id) {
+            //unregister stream
+            std::mem::drop(stream_lis);
+        } else {
+            println!("stream listener not found for the given stream_id");
+        };
+
+        if stream_states[stream_id as usize] == VIRTIO_SND_R_PCM_PREPARE {
```

Comment:
I think the stream shall be set here in the VIRTIO_SND_R_PCM_RELEASE state. Then, in prepare(), if the stream is in RELEASE, the stream shall be created and connected again.

Reply:
if the stream state is in Release, it disconnects and unregisters the stream, no destruction is done yet because of some issue upstream, so if a prepare is encountered in a Release(which I think is the norm), set state to set_params which ensures that the stream is created and connected again. I am not sure I understand the OR case you are refering to, do you mind to explain? thanks.

Reply:
We can keep it like this for now but I think we shall review the PCM state-machine at some point to be conform to the spec.